### ISSUE TITLE
nyx_mode: [QEMU-NYX] Error: ToPA allocation failure. Check kernel logs.

### ISSUE THREAD

**Issue by Rekisholm** [AFLplusplus/AFLplusplus#2249]
Error in Fuzzing libxml2 with AFL++ in Nyx-mode

Process:
1. Installed KVM-Nyx:
cd /tmp/ &&
wget -c https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.10.75/amd64/linux-image-unsigned-5.10.75-051075-generic_5.10.75-051075.202110201038_amd64.deb &&
wget -c https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.10.75/amd64/linux-modules-5.10.75-051075-generic_5.10.75-051075.202110201038_amd64.deb &&
sudo dpkg -i *.deb
sudo reboot

unzip kvm-nyx-5.10.73-1.1.zip
cd kvm-nyx-5.10.73-1.1
sudo rmmod kvm_intel kvm
sudo insmod ./kvm.ko;
sudo insmod ./kvm-intel.ko;
sudo chmod 777 /dev/kvm

2. Installed AFL++ and enabled nyx_mode:
git clone https://github.com/AFLplusplus/AFLplusplus
cd AFLplusplus
make distrib
sudo make install

3. Fuzzing libxml2 with AFL++ in Nyx-mode
https://github.com/AFLplusplus/AFLplusplus/blob/stable/nyx_mode/README.md#fuzzing-libxml2-with-afl-in-nyx-mode
./afl-fuzz -i /tmp/in/ -o /tmp/out -X /tmp/nyx_libxml2/

Then conmes to [QEMU-NYX] Error: ToPA allocation failure. Check kernel logs.
![image](https://github.com/user-attachments/assets/86c2e894-6c6e-4df8-8471-887eb9ef9294)

**Comment by vanhauser-thc** [comment]
@schumilo do you know what the issue is? he is root, the kernel modules seem to be loaded ...

**Comment by schumilo** [comment]
@Rekisholm  Could you please share a bit more information, like the specs of your machine (CPU and RAM) and OS as well as kernel version? 
An excerpt from `dmesg`would also be quite helpful.  
Thanks!

**Comment by Rekisholm** [comment]
@schumilo Thank you so much for your response! 
I investigated the issue and discovered that my Linux server was a virtual machine created from a server cluster. As a result, even though I enabled the host CPU and nested virtualization, Intel PT was still not supported on my Linux server. I have since switched to a physical machine that supports Intel PT, and QEMU-Nyx is now working perfectly. 
Thank you again!
